opsgenie/data_source_opsgenie_service_test.go

```diff
@@ -0,0 +1,88 @@
+package opsgenie
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccDataSourceOpsGenieService_Basic(t *testing.T) {
+	randomUserName := acctest.RandString(6)
+	randomTeamName := acctest.RandString(6)
+	randomServiceName := acctest.RandString(6)
+
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceOpsGenieServiceConfig(randomUserName, randomTeamName, randomServiceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataSourceOpsGenieService("opsgenie_service.test", "data.opsgenie_service.existingservice"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceOpsGenieService(src, n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		srcR := s.RootModule().Resources[src]
+		srcA := srcR.Primary.Attributes
+
+		r := s.RootModule().Resources[n]
+		a := r.Primary.Attributes
+
+		if a["id"] == "" {
+			return fmt.Errorf("Expected to get a team ID from OpsGenie")
+		}
+
+		testAtts := []string{"name", "team_id", "description"}
+
+		for _, att := range testAtts {
+			if a[att] != srcA[att] {
+				return fmt.Errorf("Expected the team %s to be: %s, but got: %s", att, srcA[att], a[att])
+			}
+		}
+
+		return nil
+	}
+}
+
+func testAccDataSourceOpsGenieServiceConfig(randomUserName, randomTeamName, randomServiceName string) string {
+	return fmt.Sprintf(`
+resource "opsgenie_user" "test" {
```

Comment:
why do we need users?

Reply:
Just added it to the team, it is not required though.

Reply:
can we delete those since that's not required

Reply:
Done